assets/javascripts/discourse/initializers/encrypt-composer.js

```diff
@@ -57,7 +57,7 @@ export default {
             // (serialization), this method is called and `isEncrypted` is
             // reset.
             if (!this.isEncryptedChanged) {
-              isEncrypted = this.siteSettings.encrypt_pms_default;
+              isEncrypted = currentUser.user_option.encrypt_pms_default;
```

Comment:
This and the code below doesn't work to set it because `currentUser.user_option` is undefined, you will need to modify the current user serializer in plugin.rb to add this to it.

Reply:
So currently the user preference is not affecting whether the lock changes when starting a new PM